online-shop/src/app/service/product.service.ts

```diff
@@ -26,7 +26,7 @@ export class ProductService{
         return this.http.put(environment.productsUrl + "/" + id, product);
     }
 
-    addProduct(product: IProduct): Observable<any>{
+    addProduct(product: any): Observable<any>{
```

Comment:
It's best to avoid using "any" types, and instead define a data type or interface that it can be used instead, as you did with `IProduct` :)